### DISCUSSION TITLE
How to setup 2 React apps and Backend in my monorepo

### DISCUSSION THREAD

**Discussion by Faizan711** [vercel/turborepo#5658] [Help] [answered]
Hello everyone, I want to create a monorepo which will have 2 react-apps for frontend of user and admin and a Backend Express server, I have cloned the react-starter app but how do i create the backend app inside the project???

**Comment by dagilleland** [comment]
What's nice about this question that @Faizan711 raises is that it speaks for those of us who are just getting into setting up a monorepo with turbo. Starter kits are good, but there are some "gaps" in the documentation and issues/discussions that appear in the official docs and on GitHub. (Kinda like when your Grandma shares a recipe but forgets to mention an ingredient - it just doesn't taste the same. 🍪 ) And those kinds of gaps in knowledge are things that it would be great to fill in!

For example:

```gherkin
As a Developer
I want to Create a new app/package to fit into my Turbo-based monorepo
So that I can Separate my monolith into component pieces
```

In practical terms, that can turn into questions like:

- Assume I want to create a new Vite/Svelte/TypeScript component library in my `~\packages\` folder.
  - [ ] Do I just do the standard `pnpm create vite@latest` inside of my `~\packages\` folder?
  - [ ] What (if anything) would I need to do with the `turbo` CLI to get it "hooked into" my build pipeline to "play nice" with the other projects?
  - [ ] Are there any important `xyz.config.js/ts` "gotacha's" that I need to watch out for?
  - [ ] How do I then "consume" or reference a component from my new library in another project? *(e.g.: I know the `name` in the `package.json` is important here)*

Answering these kinds of questions is what can give newcomers to TurboRepo (and the whole Vercel ecosystem) an opportunity to get onboard. It's a bit hard to "ramp up" when the incline makes the ramp look more like a vertical wall.

Granted, it's hard to pick a starting point of base knowledge that we might expect people to have. And documentation is *hard* (especially with the pace of change now-a-days across countless frameworks) - "document-rot" is a real thing! But, whenever someone in the community can take the time to write out a good explanation, it's impact is like doling out hard cash to the community! Priceless!!

*(I speak as a veteran instructor in computer programming at the post-secondary level - it's quite hard to condense years worth of understanding and "it depends" into a 2-hour lecture. But when I take the time to synthesize it all down, it sure helps.)*

**Reply by dagilleland** [reply]
... and ...

after posting this I notice the pinned discussion titled "What's the process for adding a new package/project?" (#243). **The answer is:** 

# [Code Shaper](https://www.code-shaper.dev/)

I think I need to save up some money and dole it out to @nareshbhatia for his EXCELLENT work on creating both documentation and a helpful app. I've only just skimmed it so far, but I gotta say that I'm impressed! Just remember to take his advice:

![image](https://github.com/vercel/turbo/assets/2947017/5d6f8611-467e-445b-975e-3a48a1812662)

**Reply by nareshbhatia** [reply]
Thank you, @dagilleland. Code Shaper is the result of months of craftsmanship and experience building production grade applications at various organizations. Glad you are enjoying it. Feedback & contributions welcome!

**Reply by nareshbhatia** [reply, marked as answer]
@Faizan711, while the Code Shaper Getting Started does not talk about setting up a back-end, the [Movie Magic](https://github.com/code-shaper/movie-magic) sample app addresses your exact use case. Look under `apps/movie-magic-api`.

**Reply by Faizan711** [reply]
Thanks for the guidance sir
